retry system calls on EINTR

dashea opened the pull request:
This issue was discovered in https://bugzilla.redhat.com/show_bug.cgi?id=1230773, tuned was crashing after waking up from hibernate due to an exception in a pyudev poll. I solved this more or less the same way it was solved in anaconda: copy what subprocess did and then add some extra bits for all the annoying differences between modules.

mulkieran commented on pyudev/_util.py:
```diff
@@ -141,3 +142,34 @@ def get_device_type(filename):
+    Handle interruptions to an interruptible system call.
+
+    Run an interruptible system call in a loop and retry if it raises EINTR.
+    The signal calls that my raise EINTR prior to Python 3.5 are listed in
```
"my" should be "may" or "might".

mulkieran commented on pyudev/_util.py:
```diff
@@ -141,3 +142,34 @@ def get_device_type(filename):
+    PEP 0475.  Any calls to these functions must be wrapped in eintr_retry_call
+    in order to handle EINTR returns in older versions of Python.
+
+    This function is based on _eintr_retry_call in python's subprocess.py.
```
This should work just as well under Python3.5 as under previous versions, since try block should always succeed immediately. Could you put a note to that effect somewhere in the docstring?

mulkieran commented on tests/test_util.py:
```diff
@@ -137,3 +137,35 @@ def test_get_device_type_not_existing(tmpdir):
+    finally:
+        os.close(fd)
+        signal.signal(signal.SIGALRM, orig_alarm)
+
```
Can you fiddle your testing so that it covers all Python versions? Perhaps add a new test for version >= 3.5 or else modify this test somehow.

Also, the convention for skipif is to make a str of the condition, like: str('sys.version_info >= (3,5)'). The seemingly redundant conversion is due to the future import of unicode strings.

mulkieran commented on pyudev/monitor.py:
```diff
@@ -34,7 +34,7 @@
 from functools import partial
 
-from pyudev._util import ensure_byte_string
+from pyudev._util import ensure_byte_string, eintr_retry_call
```
Please put all new imports on a separate line, alphabetically. This is just my aesthetic preference.

mulkieran commented:
Could you rebase changes against master as well? The plan is to put this in as a bug fix release on master branch, and then to try to move on to actual new development.

mulkieran commented:
python 3.5 is a bit too bleeding edge to make it worth testing against in this context.

No point in updating .travis.yml file.

But it still makes sense to add a dummy place-holder test, since 3.5 is on its way.

The Travis tests do not all pass, but that may be an artifact of the testing.

dashea commented:
Updated the comments and the import. Removed the skipIf from the test and moved the conditional to the block that looks for an error being raised. The rest of the test should have the same result on Python >= 3.5, so now it runs on Python >= 3.5.

mulkieran commented:
The test fails on 3.2 only. The error could be an even deeper Python error specific to 3.2 or it could be a Travis CI error. I wouldn't object skipping the test for 3.2 in some way up to just throwing out all 3.2 tests entirely as they are there as an artifact of the fact that that was the python3 in 2013. Your call.

dashea commented:
Apparently the correct way to access the members of select.error is through "args". Works with python 2.7 and 3.2.
